lib/rules/jsx-no-invalid-rel.js

```diff
@@ -0,0 +1,192 @@
+
+function splitIntoRangedParts(node, regex) {
+  const res = [];
+  const valueRangeStart = node.range[0] + 1; // the plus one is for the initial quote
+  let match;
+
+  // eslint-disable-next-line no-cond-assign
+  while ((match = regex.exec(node.value)) !== null) {
+    const start = match.index + valueRangeStart;
+    const end = start + match[0].length;
+    res.push({
+      reportingValue: `"${match[1]}"`,
+      value: match[1],
+      range: [start, end]
+    });
+  }
+
+  return res;
+}
+
+function checkLiteralValueNode(context, node, parentNodeName) {
+  if (typeof node.value !== 'string') {
+    return context.report({
+      node,
+      message: '"rel" attribute only supports strings.',
+      fix(fixer) {
+        return fixer.remove(node.parent.parent);
+      }
+    });
+  }
+
+  if (!node.value.trim()) {
+    return context.report({
+      node,
+      message: 'An empty "rel" attribute is meaningless.',
+      fix(fixer) {
+        return fixer.remove(node);
+      }
+    });
+  }
+
+  const parts = splitIntoRangedParts(node, /([^\s]+)/g);
+  for (const part of parts) {
+    const allowedTags = standardValues.get(part.value);
+    if (!allowedTags) {
+      context.report({
+        node,
+        message: `${part.reportingValue} is never a valid "rel" attribute value.`,
+        fix(fixer) {
+          return fixer.removeRange(part.range);
+        }
+      });
+    } else if (!allowedTags.has(parentNodeName)) {
+      context.report({
+        node,
+        message: `${part.reportingValue} is not a valid "rel" attribute value for <${parentNodeName}>.`,
+        fix(fixer) {
+          return fixer.removeRange(part.range);
+        }
+      });
+    }
+  }
+
+  const whitespaceParts = splitIntoRangedParts(node, /(\s+)/g);
+  for (const whitespacePart of whitespaceParts) {
+    if (whitespacePart.value !== ' ' || whitespacePart.range[0] === (node.range[0] + 1) || whitespacePart.range[1] === (node.range[1] - 1)) {
+      context.report({
+        node,
+        message: '"rel" attribute values should be space delimited.',
+        fix(fixer) {
+          return fixer.removeRange(whitespacePart.range);
+        }
+      });
+    }
+  }
+}
+
+module.exports = {
+  meta: {
+    fixable: 'code',
+    docs: {
+      description: 'Forbid `rel` attribute with an invalid value`',
+      category: 'Possible Errors',
+      url: docsUrl('jsx-no-invalid-rel')
+    }
+  },
+
+  create(context) {
+    return {
+      JSXAttribute(node) {
+        // ignore attributes that aren't "rel"
+        if (node.name.name !== 'rel') {
+          return;
+        }
+
+        const parentNodeName = node.parent.name.name;
+        if (!components.has(parentNodeName)) {
+          return context.report({
+            node,
+            message: 'The "rel" attribute only has meaning on `<link>`, `<a>`, `<area>`, and `<form>` tags.',
+            fix(fixer) {
+              return fixer.remove(node);
+            }
```

Comment:
this fixer function can be defined at module level and reused in a bunch of places.